roles/translator/src/lib/sv2/channel_manager/message_handler.rs

```diff
@@ -0,0 +1,426 @@
+                    return new_open_extended_mining_channel_success.into_static();
+                }
+            }
+
+            // If we are not in aggregated mode, we just insert the extended channel into the map
+            channel_manager_data.extended_channels
+                .insert(m.channel_id, Arc::new(RwLock::new(extended_channel)));
+
+            m.into_static()
+        }).unwrap();
+
+        self.channel_state
+            .sv1_server_sender
+            .send(Mining::OpenExtendedMiningChannelSuccess(success.clone()))
+            .await
+            .map_err(|e| {
+                error!("Failed to send OpenExtendedMiningChannelSuccess: {:?}", e);
+                HandlerError::External(Box::new(TproxyError::ChannelErrorSender))
+            })?;
+
+        Ok(())
+    }
+
+    async fn handle_open_mining_channel_error(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::OpenMiningChannelError<'_>,
+    ) -> Result<(), HandlerError> {
+        error!(
+            "Received OpenExtendedMiningChannelError from upstream with error code {}",
+            std::str::from_utf8(m.error_code.as_ref()).unwrap_or("unknown error code")
+        );
+        todo!("OpenExtendedMiningChannelError not handled yet");
+    }
+
+    async fn handle_update_channel_error(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::UpdateChannelError<'_>,
+    ) -> Result<(), HandlerError> {
+        error!(
+            "Received UpdateChannelError from upstream with error code {}",
+            std::str::from_utf8(m.error_code.as_ref()).unwrap_or("unknown error code")
+        );
+        Ok(())
+    }
+
+    async fn handle_close_channel(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::CloseChannel<'_>,
+    ) -> Result<(), HandlerError> {
+        info!(
+            "Received CloseChannel from upstream for channel id: {}",
+            m.channel_id
+        );
+        _ = self.channel_manager_data.safe_lock(|channel_data_manager| {
+            if channel_data_manager.mode == ChannelMode::Aggregated {
+                if channel_data_manager.upstream_extended_channel.is_some() {
+                    channel_data_manager.upstream_extended_channel = None;
+                }
+            } else {
+                channel_data_manager.extended_channels.remove(&m.channel_id);
+            }
+        });
+        Ok(())
+    }
+
+    async fn handle_set_extranonce_prefix(
+        &mut self,
+        _m: roles_logic_sv2::mining_sv2::SetExtranoncePrefix<'_>,
+    ) -> Result<(), HandlerError> {
+        unreachable!("Cannot process SetExtranoncePrefix since set_extranonce is not supported for majority of sv1 clients");
+    }
+
+    async fn handle_submit_shares_success(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::SubmitSharesSuccess,
+    ) -> Result<(), HandlerError> {
+        info!(
+            "Received SubmitSharesSuccess from upstream for channel id: {} ✅",
+            m.channel_id
```

Comment:
here, we would only be showing `channel_id`, but not other relevant message fields

we have `Display` trait implementations for all messages now, we should leverage them
```suggestion
            "Received: {}", m
```

Reply:
and the `debug!` print below could be removed